src/lib.rs

```diff
@@ -64,8 +64,6 @@ where
 pub enum Orientation {
     Portrait = 0b0000_0000,         // no inverting
     Landscape = 0b0110_0000,        // invert column and page/column order
```

Comment:
I would also be setting this to just `0b0010_0000` as the 2nd bit can be inverted if required.

Reply:
I'm wondering if it wouldn't be more straightforward to have a flat enum for all the possible combinations rather than the "invert x/y".

Reply:
That is going to be a lot of combinations. Basically there are 3 bits, so 2*2*2 = 8 combinations. Think it might be more confusing, not less confusing.

Although personally, wondering if it would be better just to have:

```
pub enum Orientation {
    Portrait,
    Landscape,
}
```

And have the `set_orientation` function map that to the correct value.

Reply:
Also worth noting, yes, we really do need to support all of 8 combinations. Depending on how the display is mounted and if the hardware reverses the image or not.

Reply:
Hmm the main issue I have with this is that splitting the orientation enum and the "revert" values seem confusing. We could do something like
```rust
pub enum Orientation {
    Portrait(bool, bool), // x/y invert
    Landscape(bool, bool), // x/y invert
}
```

Or possibly introduce a new sub-enum for "inversion" and use that as the inner value.

Reply:
Hmmm. Not sure I can see how this could be less confusing. Also most of the code only needs to know if it is Portrait/Landscape. If the display is inverted or not makes no difference. i.e. the invert parameters are only really needed to configure the MADCTL register and don't matter afterward. In fact the invert parameters have no real meaning outside the MADCTL because there is no real way to know what the hardware is doing with them. Maybe setting invert_x will invert the display, or maybe setting invert_x will restore the display.

Maybe that is the problem though. We use the `set_orientation` to set parameters that are not orientation. You wouldn't set the RGB/BGR value here. But the problem being it is interconnected. When setting orientation you typically will (always) have to adjust the invert values at the same time. Which is why we do it in the `set_orientation` call.